particle-iot/device-os
# Feature/ch1424/explicit publish scope

### Problem

Not presently possible to capture and verify expected compiler warnings/errors, such as deprecated or removed APIs. 

### Solution

Some additional cucumber steps have been added to capture and verify the compiler output.

This PR also includes changes to the build which allow the current directory to be outside of the build tree. 

### Steps to Test

The [explicit_publish_scope](https://github.com/spark/firmware/compare/feature%2Fch1424%2Fexplicit_publish_scope?expand=1#diff-483d574f41c5ee47b0560ff99dc8fd2f) gherkin test uses the new tests to verify that attempts to call removed APIs result in a compiler error. 

---

### Completeness
